radis/lbl/factory.py

```diff
@@ -1196,6 +1494,70 @@ def non_eq_spectrum(
             self._clean_temp_file()
             raise
 
+    def _get_log_2gs(self):
+        """ Returns log_2gs if it already exists in the dataframe, otherwise computes it using gamma_air """
+        df = self.df0
+
+        # if the column already exists, then return
+        if "log_2gs" in df.columns:
+            return df["log_2gs"]
+
+        try:
+            gamma_air = df["airbrd"].to_numpy()
+            log_2gs = np.log(2 * gamma_air)
+            df["log_2gs"] = log_2gs
+            return log_2gs
+        except KeyError:
+            raise KeyError(
+                "Cannot find air-broadened half-width or log_2gs in the database... please check the database"
+            )
```

Comment:
Tip : this allows you to still get the proper Traceback (new raised error will inherit from the previous)
```suggestion
        except KeyError as err:
            raise KeyError(
                "Cannot find air-broadened half-width or log_2gs in the database... please check the database"
            ) from err
```